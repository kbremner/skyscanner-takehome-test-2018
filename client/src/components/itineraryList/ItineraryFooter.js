import React from 'react';
import BpkText from 'bpk-component-text';
import BpkButton from 'bpk-component-button';
import PropTypes from '../../types';
import './ItineraryFooter.scss';

const ItineraryFooter = ({ className, price, agentName, bookingLink }) => (
  <div className={[className, 'itinerary-footer'].join(' ')}>
      <div className="itinerary-footer__quote">
        <BpkText className="itinerary-footer__price" textStyle="xl">{price}</BpkText>
        <BpkText className="itinerary-footer__agent">{agentName}</BpkText>
      </div>
      <BpkButton href={bookingLink} target="_blank">Select</BpkButton>
    </div>
);

ItineraryFooter.propTypes = {
  price: PropTypes.string.isRequired,
  agentName: PropTypes.string.isRequired,
  bookingLink: PropTypes.string.isRequired,
  className: PropTypes.string
};

ItineraryFooter.defaultProps = {
  className: null
};

export default ItineraryFooter;
