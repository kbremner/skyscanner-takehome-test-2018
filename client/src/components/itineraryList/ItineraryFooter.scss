@import "~bpk-mixins";

.itinerary-footer {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;

  .itinerary-footer__quote {
    display: flex;
    flex-direction: column;
    text-align: left;
  }
  
  .itinerary-footer__agent {
    color: $bpk-color-gray-300;
  }

  .itinerary-footer__agent {
    line-height: $bpk-line-height-sm;
  }

  .itinerary-footer__price {
    line-height: $bpk-line-height-base;
  }
}