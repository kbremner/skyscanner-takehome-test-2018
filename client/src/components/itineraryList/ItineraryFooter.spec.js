import React from 'react';
import { shallow } from 'enzyme';
import ItineraryFooter from './ItineraryFooter';

const defaultProps = {
  className: null,
  price: '$234.45',
  agentName: 'test agent',
  bookingLink: 'http://example.com'
};

const setup = (overrides = {}) => {
  const props = {
    ...defaultProps,
    ...overrides
  };

  const wrapper = shallow(<ItineraryFooter {...props} />);

  return { wrapper, props };
};

describe('ItineraryFooter', () => {
  it('renders', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('adds class name to card provided', () => {
    const { wrapper, props } = setup({ className: 'test-class-name' });
    expect(wrapper.hasClass(props.className)).toBe(true);
  });
});
