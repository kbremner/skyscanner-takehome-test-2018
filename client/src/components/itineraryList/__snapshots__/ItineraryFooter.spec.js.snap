// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryFooter renders 1`] = `
<div
  className=" itinerary-footer"
>
  <div
    className="itinerary-footer__quote"
  >
    <BpkText
      bold={false}
      className="itinerary-footer__price"
      tagName="span"
      textStyle="xl"
    >
      $234.45
    </BpkText>
    <BpkText
      bold={false}
      className="itinerary-footer__agent"
      tagName="span"
      textStyle="base"
    >
      test agent
    </BpkText>
  </div>
  <BpkButton
    className={null}
    destructive={false}
    disabled={false}
    featured={false}
    href="http://example.com"
    iconOnly={false}
    large={false}
    link={false}
    onClick={null}
    secondary={false}
    submit={false}
    target="_blank"
  >
    Select
  </BpkButton>
</div>
`;
